modules/csvparser/csvparser.c

```diff
@@ -139,8 +139,8 @@ void
 csv_parser_set_drop_invalid(LogParser *s, gboolean drop_invalid)
 {
   CSVParser *self = (CSVParser *) s;
-
-  self->drop_invalid = drop_invalid;
+  if (drop_invalid)
+    self->on_error |= ON_ERROR_DROP_MESSAGE;
```

Comment:
This won't work well in the case of
```
drop-invalid(yes)
drop-invalid(no)
```
because we never clear the `ON_ERROR_DROP_MESSAGE` bitfield.